hal/src/main/native/mau/AnalogAccumulator.cpp

```diff
@@ -0,0 +1,121 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "HAL/AnalogAccumulator.h"
+
+#include "AnalogInternal.h"
+#include <VMXIO.h>
+#include <VMXErrors.h>
+#include <VMXResource.h>
+#include "VMXPointers.h"
+
+using namespace hal;
+
+namespace hal {
+    namespace init {
+        void InitializeAnalogAccumulator() {}
+    }
+}
+
+extern "C" {
+    HAL_Bool HAL_IsAccumulatorChannel(HAL_AnalogInputHandle analogPortHandle, int32_t* status) {
+        auto port = analogInputHandles->Get(analogPortHandle);
+        if (port == nullptr) {
+            *status = HAL_HANDLE_ERROR;
+            return false;
+        }
+        for (int32_t i = 0; i < kNumAccumulators; i++) {
+            if (port->channel == kAccumulatorChannels[i]) return true;
+        }
+        return false;
+    }
+
+    void HAL_InitAccumulator(HAL_AnalogInputHandle analogPortHandle, int32_t* status) {
+        auto port = analogInputHandles->Get(analogPortHandle);
+        if (port == nullptr) {
+            //    *status = HAL_HANDLE_ERROR;
+            return;
+        }
+
+        if (!HAL_IsAccumulatorChannel(analogPortHandle, status)) {
+            *status = HAL_INVALID_ACCUMULATOR_CHANNEL;
+            return;
+        }
+
+        //  SimAnalogInData[port->channel].SetAccumulatorInitialized(true);
```

Comment:
Here's the code from the VMX-pi C++ analog_io example, which allocates an accumulator resource.

if (vmx.io.ActivateSinglechannelResource(analog_in_chan_index, VMXChannelCapability::AccumulatorInput,
          accumulator_res_handles[accum_res_index], &accum_config, &vmxerr)) {